scripts/changeset.sh

```
@@ -0,0 +1,332 @@
+    echo "$files" | while read -r file; do
+        if [ -n "$file" ]; then
+            # Extract extension (handle files without extension)
+            ext="${file##*.}"
+            if [ "$ext" = "$file" ] || [ -z "$ext" ]; then
+                echo "no-ext"
+            else
+                echo "$ext"
+            fi
+        fi
+    done | sort | uniq -c | sort -rn | while read -r count ext; do
+        local name=$(get_extension_name "$ext")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)